### DISCUSSION TITLE
Cannot find module '@@/plugin-dva/exports' or its corresponding type declarations.ts(2307)

### DISCUSSION THREAD

**Discussion by Khadhiri786** [umijs/umi#11900] [General]
import { useSelector } from '@@/plugin-dva/exports';

I'm getting error while upgrading umijs v3 to umijs v4 .how to fix this
`  The plugin "esbuildAliasPlugin" was triggered by this import`

**Comment by fz6m** [comment]
In umi 4 , you can import all dva methods from `umi` or `@@/plugin-dva` :

```ts
import { connect, useDispatch, useStore, useSelector, getDvaApp } from 'umi' // or '@@/plugin-dva'

// See `./src/.umi/plugin-dva`
```

**Reply by Khadhiri786** [reply]
I have doubt while upgrading umijs from v3 to v4, 
Removed package
-umi,@umijs/preset-react
upgraded package
"react-dom": "^18.2.0"
"react": "^18.2.0"
"@umijs/max": "^4.0.88"
I'm getting error
`error - MFSU latest build error:  AssertionError [ERR_ASSERTION]: dependence not found: @umijs/plugin-antd/node_modules/antd/lib/style/themes/default.less
`
How to fix this error

**Reply by fz6m** [reply]
`@umijs/plugin-xxx` is umi 3 plugins , not for umi v4. plz remove all `@umijs/plugin-xxx` plugins.

```diff
- plugins: ['@umijs/plugin-antd']
+ antd: {}
```

the antd plugin is included in max (umi 4) by default.

if you are using antd 4, should be manually install `antd: "4.x.x"` to package.json, otherwise will use antd 5 by default.

**Reply by tnsivanagarajan** [reply]
> In umi 4 , you can import all dva methods from `umi` or `@@/plugin-dva` :
> 
> ```ts
> import { connect, useDispatch, useStore, useSelector, getDvaApp } from 'umi' // or '@@/plugin-dva'
> 
> // See `./src/.umi/plugin-dva`
> ```

I'm getting this below error.

Module '"umi"' has no exported member 'useDispatch'.
